cli/azd/pkg/tools/powershell/powershell.go

```diff
@@ -25,9 +28,21 @@ type powershellScript struct {
 	envVars       []string
 }
 
+func checkInstalled(options tools.ExecOptions) (err error) {
+	return tools.ToolInPath(options.UserPwsh)
```

Comment:
With the recent changes in #4595, I think you may have to do `string.Split(options.UserPwsh, " ")[0]` to handle `pwsh -NoProfile` for example.